### PR TITLE
Single key option for Slicer and doc improvements

### PR DESCRIPTION
**Pull request by SvenDS9**
Single key option for Slicer and doc improvements

### Changes

- Enable Slicer to also work for a single key + functional test
- Fix typos in doc
- Add laion-example to examples page

### CODE REVIEW

File: torchdata/datapipes/iter/transform/callable.py [hunk 1]
```diff
@@ -285,6 +295,8 @@ def __iter__(self) -> Iterator[T_co]:
             elif isinstance(old_item, dict):
                 if isinstance(self.index, list):
                     new_item = {k: v for (k, v) in old_item.items() if k in self.index}  # type: ignore[assignment]
+                elif self.index in old_item.keys():
+                    new_item = {self.index: old_item.get(self.index)}  # type: ignore[assignment]
```
Comment (ejguan) [inline torchdata/datapipes/iter/transform/callable.py:299]: TBH, dict is weird for `slice`.

**Review by ejguan — APPROVED** [review]
LGTM, thank you

File: docs/source/examples.rst [hunk 2]
```diff
@@ -75,6 +75,13 @@ semantic classes. Here is a
 <https://github.com/tcapelle/torchdata/blob/main/01_Camvid_segmentation_with_datapipes.ipynb>`_
 created by our community.
 
+laion2B-en-joined
+^^^^^^^^^^^^^^^^^^^^^^
+The `laion2B-en-joined dataset <https://huggingface.co/datasets/laion/laion2B-en-joined>`_ is a subset of the `LAION-5B dataset <https://laion.ai/blog/laion-5b/>`_ containing english captions, URls pointing to images,
+and  other metadata. It contains around 2.32 billion entries.
```
Comment (NivekT) [inline docs/source/examples.rst:81]:
```suggestion
and other metadata. It contains around 2.32 billion entries.
```

File: docs/source/reading_service.rst [hunk 3]
```diff
@@ -11,11 +11,11 @@ Features
 Dynamic Sharding
 ^^^^^^^^^^^^^^^^
 
-Dynamic sharding is achieved by ``MultiProcessingReadingService`` and ``DistributedReadingService`` to shard the pipeline based on the information of corresponding multiprocessing and distributed workers. And, TorchData offers two types of ``DataPipe`` letting users to define the sharding place within the pipeline.
+Dynamic sharding is achieved by ``MultiProcessingReadingService`` and ``DistributedReadingService`` to shard the pipeline based on the information of corresponding multiprocessing and distributed workers. And, TorchData offers two types of ``DataPipe`` letting users define the sharding place within the pipeline.
 
-- ``sharding_filter``: When the pipeline is replicable, each distributed/multiprocessing worker loads data from one replica of the ``DataPipe`` graph, and skip the data not blonged to the corresponding worker at the place of ``sharding_filter``.
+- ``sharding_filter``: When the pipeline is replicable, each distributed/multiprocessing worker loads data from one replica of the ``DataPipe`` graph, and skips the data not belonging to the corresponding worker at the place of ``sharding_filter``.
```
Comment (NivekT) [inline docs/source/reading_service.rst:16]: FYI, I have a fix to this in a separate PR soon to land. Rebasing will be necessary

**Pull request closed without merging**